core/src/tests/recent_job_results_list.cc

```diff
@@ -48,20 +48,50 @@ static bool StateFileExists(const char* dir, const char* name, int port)
   return stat(path.c_str(), &buf) == 0;
 }
 
+static bool CopyTestStateFileFromOriginal(std::string orig_path,
```

Comment:
As we now have c++17 available this should be easier with 
std::filesystem::copy

Reply:
the filesystem interface is not really usable today. A lot of platforms require to link with a special lib and some don't support the interface yet.